libraries/AP_SmartAudio/AP_SmartAudio.cpp

```diff
@@ -311,7 +308,7 @@ bool AP_SmartAudio::update(bool force)
 
     // sync power with AP_VideoTx backend
     if(sync_power()){
-        set_power_mw(AP::vtx().get_power_mw(),_vtx_current_state->version);
+        set_power(AP::vtx().get_configured_power_mw(), AP::vtx().get_configured_power_dbm());
```

Comment:
When syncing power the provided values are the configured values, so any other change into de the vtx backend not persisted in config will be ignored, and not reflected into the hw_vtx

Reply:
Again this is a simplification - you are replicating a bunch of code that is already in AP_VideoTX. I understand what you are trying to do, but this is why update_power() and friends exist.

Reply:
Message received :), i considered the updated_x method redundant because this checking is vs sma_vtx not taking in account the configured values, which are checked into the hotdeploy. I'll rewrite the logic for this.